PokeOneWeb.DataSync.Tests/GoogleSpreadsheet/Import/Impl/Sheets/Pokemon/PokemonSheetRowParserTest.cs

```diff
@@ -0,0 +1,444 @@
+                0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0",
+                0, 0, 0, 0, 0, 0,
+                0, 0, 0, 0, 0, 0,
+                "0", "", "0", "", "", "", false, false, 0, 0, false, 0.5, 0, 0, 0, 0,
+                "", "", "", "", "", "", "", "excessive value"
+            };
+
+            // Act
+            var actual = () => parser.ReadRow(values);
+
+            // Assert
+            actual.Should().Throw<InvalidRowDataException>();
+        }
+
+        [Fact]
+        public void ReadRow_WithValuesNull_ShouldThrow()
+        {
+            // Arrange
+            var parser = new PokemonSheetRowParser();
+            List<object> values = null;
+
+            // Act
+            var actual = () => parser.ReadRow(values);
+
+            // Assert
+            actual.Should().Throw<InvalidRowDataException>();
+        }
+
+        [Theory]
+        [InlineData("")] // SortIndex must be int
+        [InlineData(0, "")] // PokedexNumber must be int
+        [InlineData(0, 0, "")] // PokemonSpeciesName must be non-empty
+        [InlineData(0, 0, "0", "")] // DefaultVarietyName must be non-empty
+        [InlineData(0, 0, "0", "0", "")] // PokemonVarietyName must be non-empty
+        [InlineData(0, 0, "0", "0", "0", "")] // ResourceName must be non-empty
+        [InlineData(0, 0, "0", "0", "0", "0", "")] // PokemonFormName must be non-empty
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "")] // AvailabilityName must be non-empty
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "")] // SpriteName must be non-empty
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", "")] // DoInclude must be boolean
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "")] // DefaultFormName must be non-empty
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", "")] // Attack must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, "")] // SpecialAttack must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, "")] // Defense must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, "")] // SpecialDefense must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, "")] // Speed must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, "")] // HitPoints must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, "")] // AttackEvYield must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, 0, "")] // SpecialAttackEvYield must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, 0, 0, "")] // DefenseEvYield must be int
+        [InlineData(0, 0, "0", "0", "0", "0", "0", "0", "0", false, "0", 0, 0, 0, 0, 0, 0, 0, 0, 0, "")] // SpecialDefenseEvYield must be int
```

Comment:
Damn son :0

Reply:
Yeah its ugly but follows the same principle as all other tests of this kind. Not entirely happy with it either, but havent come up with another solution that I am satisfied with.